openvpn/middlewares/client/auth/middleware.go

```diff
@@ -7,25 +7,28 @@ import (
 	"regexp"
 )
 
+// CredentialsProvider returns client's current auth primitives (i.e. customer identity signature / node's sessionId)
+type CredentialsProvider func() (username string, password string, err error)
```

Comment:
`username, password` pair is strongly coupled - it's being passed around the system in many places and we have quite a few methods which returns `(username, password string, err error)`.
Maybe it's time to merge them into struct?
```go
type Credentials struct {
    username, password string
}
```

Reply:
I.e. we wouldn't have to explain what is "auth primitives" in many places, as we do now - we could do that in one place.

Reply:
Yeah that's true. but since our middleware packageing is a bit different, we don't have a possibility to share credentials structure between client and server. But good point and it can be improved with separate PR.